alchemy/src/cloudflare/warp-default-profile.ts

```diff
@@ -0,0 +1,322 @@
+ *
+ * const defaultProfile = await WarpDefaultProfile("default", {
+ *   serviceModeV2: { mode: "warp" },
+ *   autoConnect: 0,
+ *   captivePortal: 180,
+ *   supportUrl: "https://support.example.com"
+ * });
+ *
+ * @example
+ * ## Default profile with split tunnel
+ *
+ * Configure default split tunnel behavior for all devices
+ *
+ * const defaultWithSplitTunnel = await WarpDefaultProfile("default", {
+ *   serviceModeV2: { mode: "warp" },
+ *   splitTunnel: {
+ *     mode: "exclude",
+ *     entries: [
+ *       { address: "10.0.0.0/8", description: "Internal network" },
+ *       { address: "192.168.0.0/16", description: "Local network" }
+ *     ]
+ *   }
+ * });
+ *
+ * @example
+ * ## Locked default profile
+ *
+ * Create a locked-down default configuration
+ *
+ * const lockedDefault = await WarpDefaultProfile("default", {
+ *   serviceModeV2: { mode: "warp" },
+ *   switchLocked: true,
+ *   allowedToLeave: false,
+ *   disableAutoFallback: true,
+ *   tunnelProtocol: "wireguard"
+ * });
+ */
+export const WarpDefaultProfile = Resource(
+  "cloudflare::WarpDefaultProfile",
+  async function (
+    this: Context<WarpDefaultProfile>,
+    id: string,
+    props: WarpDefaultProfileProps = {},
+  ): Promise<WarpDefaultProfile> {
+    const api = await createCloudflareApi(props);
+
+    if (this.phase === "delete") {
+      // Default profile cannot be deleted, just remove from state
+      return this.destroy();
+    }
+
+    // Update default profile (single PATCH endpoint)
+    await updateDefaultPolicy(api, props);
+
+    return {
+      ...props,
+      modifiedAt: Date.now(),
+    };
+  },
+);
+
+/**
+ * Internal API response type for default policy
+ * @internal
+ */
+interface CloudflareDefaultPolicyResponse {
+  device_settings?: {
+    service_mode_v2?: {
+      mode: string;
+      port?: number;
+    };
+    disable_auto_fallback?: boolean;
+    allow_mode_switch?: boolean;
+    switch_locked?: boolean;
+    tunnel_protocol?: string;
+    auto_connect?: number;
+    allowed_to_leave?: boolean;
+    captive_portal?: number;
+    support_url?: string;
+    exclude_office_ips?: boolean;
+    lan_allow_minutes?: number;
+    lan_allow_subnet_size?: number;
+  };
+}
+
+async function updateDefaultPolicy(
+  api: CloudflareApi,
+  props: WarpDefaultProfileProps,
+): Promise<void> {
+  // Build device settings object (includes split tunnel)
+  const deviceSettings: any = {};
+  if (props.serviceModeV2) {
+    deviceSettings.service_mode_v2 = {
+      mode: props.serviceModeV2.mode,
+      ...(props.serviceModeV2.port && { port: props.serviceModeV2.port }),
+    };
+  }
+  if (props.disableAutoFallback !== undefined) {
+    deviceSettings.disable_auto_fallback = props.disableAutoFallback;
+  }
+  if (props.allowModeSwitch !== undefined) {
+    deviceSettings.allow_mode_switch = props.allowModeSwitch;
+  }
+  if (props.switchLocked !== undefined) {
+    deviceSettings.switch_locked = props.switchLocked;
+  }
```

Comment:
I don't know if this is necessary since `undefined` is omitted when serialized as JSON?